# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/FileSystemManagerParentFactory.cpp
# Commit: cfa7bbd206c3
# Full Hash: cfa7bbd206c3141f7d0b2801586dec142f20a11e
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-08-31 21:54:47
# Regressor Bug: 1786465
# File Overlap Count: 2
# Description:
#   Bug 1786465 - Create only one FileSystemDataManager per origin; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155553
# ==============================================================================

diff -r a2f6102fc8ea -r cfa7bbd206c3 dom/fs/parent/FileSystemManagerParentFactory.cpp
--- a/dom/fs/parent/FileSystemManagerParentFactory.cpp	Wed Aug 31 09:34:55 2022 +0000
+++ b/dom/fs/parent/FileSystemManagerParentFactory.cpp	Wed Aug 31 09:34:55 2022 +0000
@@ -55,7 +55,7 @@
   // This opens the quota manager, which has to be done on PBackground
   QM_TRY_UNWRAP(
       RefPtr<fs::data::FileSystemDataManager> dataManager,
-      fs::data::FileSystemDataManager::CreateFileSystemDataManager(origin),
+      fs::data::FileSystemDataManager::GetOrCreateFileSystemDataManager(origin),
       IPC_OK(), sendBackError);
 
   nsCOMPtr<nsIThread> pbackground = NS_GetCurrentThread();